src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs

```diff
@@ -3886,6 +3886,67 @@ public void XA1018 ()
 			}
 		}
 
+		static readonly object [] XA1027XA1028Source = new object [] {
+			new object [] {
+				/* linkTool */                   "r8",
+				/* enableProguard */             null,
+				/* androidEnableProguard */      "true",
+				/* expectedBuildResult */        true,
+				/* expectedWarning */            "0 Warning(s)",
+			},
+			new object [] {
+				/* linkTool */                   "proguard",
+				/* enableProguard */             null,
+				/* androidEnableProguard */      "true",
+				/* expectedBuildResult */        false,
+				/* expectedWarning */            "0 Warning(s)",
+			},
+			new object [] {
+				/* linkTool */                   null,
+				/* enableProguard */             null,
+				/* androidEnableProguard */      null,
+				/* expectedBuildResult */        true,
+				/* expectedWarning */            "0 Warning(s)",
+			},
+			new object [] {
+				/* linkTool */                   null,
+				/* enableProguard */             "true",
+				/* androidEnableProguard */      null,
+				/* expectedBuildResult */        false,
+				/* expectedWarning */            "warning XA1027:",
+			},
+			new object [] {
+				/* linkTool */                   null,
+				/* enableProguard */             null,
+				/* androidEnableProguard */      "true",
+				/* expectedBuildResult */        false,
+				/* expectedWarning */            "warning XA1028:",
+			}
+		};
+
+		[Test]
+		[TestCaseSource (nameof (XA1027XA1028Source))]
+		public void XA1027XA1028 (string linkTool, string enableProguard, string androidEnableProguard, bool expectedBuildResult, string expectedWarning)
```

Comment:
I think this new test fails in some cases:

![image](https://user-images.githubusercontent.com/840039/95086657-5d26ca00-06e6-11eb-8aab-92d2c4564962.png)